### PR TITLE
fix(authorization): Sync roles after an application is created

### PR DESCRIPTION
**Pull request by AbdulRahmanAlHamali**
Should fix https://github.com/spinnaker/spinnaker/issues/5063

### CODE REVIEW

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy [hunk 1]
```diff
@@ -98,7 +102,9 @@ public class ApplicationsController {
   @ApiOperation(value = "", notes = "Create an application")
   @RequestMapping(method = RequestMethod.POST)
   Application create(@RequestBody final Application app) {
-    return getApplication().initialize(app).withName(app.getName()).save()
+    Application createdApplication = getApplication().initialize(app).withName(app.getName()).save()
+    fiatService.sync()
```
Comment (cfieber) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy]:
```suggestion
    try {
      fiatService.ifPresent { it.sync() }
    } catch (Exception ignored) {
      log.warn("failed to trigger fiat permission sync", ignored)
    } 
```

Comment (cfieber) [reply front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy]: we've already created the application - may as well return it

File: front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy [hunk 2]
```diff
@@ -59,6 +60,9 @@ public class ApplicationsController {
   @Autowired(required = false)
   List<ApplicationEventListener> applicationEventListeners = []
 
+  @Autowired
+  FiatService fiatService;
```
Comment (cfieber) [inline front50-web/src/main/groovy/com/netflix/spinnaker/front50/controllers/v2/ApplicationsController.groovy]:
```suggestion
  Optional<FiatService> fiatService;
```

it is possible to run front50 without fiat

**Comment by AbdulRahmanAlHamali** [comment]
Actually I think we should use [FiatStatus.isEnabled](https://github.com/spinnaker/fiat/blob/87a0d2410244f1a906a3220903aa121ab5042b24/fiat-api/src/main/java/com/netflix/spinnaker/fiat/shared/FiatStatus.java) instead of try/catch. No?

**Pull request merged**